src/garage/tf/algos/rl2.py

```diff
@@ -1,15 +1,340 @@
+        return paths['average_return']
+
+    def get_exploration_policy(self):
+        """Return a policy used before adaptation to a specific task.
+
+        Each time it is retrieved, this policy should only be evaluated in one
+        task.
+
+        Returns:
+            object: The policy used to obtain samples that are later
+                used for meta-RL adaptation.
+
+        """
+        return self._policy
+
+    def adapt_policy(self, exploration_policy, exploration_trajectories):
+        """Produce a policy adapted for a task.
+
+        Args:
+            exploration_policy (garage.Policy): A policy which was returned
+                from get_exploration_policy(), and which generated
+                exploration_trajectories by interacting with an environment.
+                The caller may not use this object after passing it into this
+                method.
+            exploration_trajectories (garage.TrajectoryBatch): Trajectories to
+                adapt to, generated by exploration_policy exploring the
+                environment.
+
+        Returns:
+            garage.Policy: A policy adapted to the task represented by the
+                exploration_trajectories.
+
+        """
+        return exploration_policy
+
+    def _process_samples(self, itr, paths):
+        # pylint: disable=too-many-statements
+        """Return processed sample data based on the collected paths.
+
+        Args:
+            itr (int): Iteration number.
+            paths (OrderedDict[dict]): A list of collected paths for each
+                task. In RL^2, there are n environments/tasks and paths in
+                each of them will be concatenated at some point and fed to
+                the policy.
+
+        Returns:
+            dict: Processed sample data, with key
+                * observations: (numpy.ndarray)
+                * actions: (numpy.ndarray)
+                * rewards: (numpy.ndarray)
+                * returns: (numpy.ndarray)
+                * valids: (numpy.ndarray)
+                * agent_infos: (dict)
+                * env_infos: (dict)
+                * paths: (list[dict])
+                * average_return: (numpy.float64)
+
+        Raises:
+            ValueError: If 'batch_idx' is not found.
+
+        """
+        concatenated_path_in_meta_batch = []
+        lengths = []
+
+        paths_by_task = collections.defaultdict(list)
+        for path in paths:
+            path['returns'] = np_tensor_utils.discount_cumsum(
+                path['rewards'], self._discount)
+            path['lengths'] = [len(path['rewards'])]
+            if 'batch_idx' in path:
+                paths_by_task[path['batch_idx']].append(path)
+            elif 'batch_idx' in path['agent_infos']:
+                paths_by_task[path['agent_infos']['batch_idx'][0]].append(path)
+            else:
+                raise ValueError('Batch idx is required for RL2 but not found')
```

Comment:
Can you mention in this error message to use `RL2Worker`? As a user, I want the error message to tell me how to fix the problem.

Reply:
fixed